lib/msf/core/post/file.rb

```diff
@@ -638,6 +642,153 @@ def copy_file(src_file, dst_file)
+  # what groups are allowed to access the file that icacls was run on, and which
+  # permissions these groups have on that file.
+  #
+  # The resulting array that is returned is a multidimensional array where each
+  # element in the array is another array containing the name of the group as
+  # its first element, followed by an array containing the permissions
+  # that the group has on the file.
+  #
+  # @param input_string [String] String containing icacls output to parse for group permissions.
+  # @return [Array] Multidimensonal array containing the groups which have access to the file and what permissions they have on the file.
+  def grab_icacls_groups_permissions(input_string)
+    final_permissions_array = []
+    icacls_groups_permissions_array = input_string.scan(/\w:(\(.*\))$/)
+    for i in 0...icacls_groups_permissions_array.length
+      icacls_groups_permissions_array[i][0].strip! # Strip any leading or trailing characters off of each of the group permissions found via the regex.
+
+      # Permissions can exist inside a () section as comma delimited entries. To solve this we first remove the ( and ) characters,
+      # then we split the list on the delimiter, aka the comma character, before finally appending the resulting array to the end of
+      # final_permissions_array.
+      final_permissions_array << icacls_groups_permissions_array[i][0].gsub(/[()]/, '').split(',')
+    end
+    final_permissions_array
+  end
+
+  #
+  # Find what groups the current user is a member of.
+  #
+  # @return [Array] Array of groups the current user is a member of.
+  def win_find_current_user_groups
+    cuser_groups = cmd_exec('whoami /groups').to_s
+    cuser_groups_array = cuser_groups.scan(/^(?:\w+ ?[\w\-]*+ ?[\w\-]*+\\?[\w\-]*+ ?[\w\-]*+ ?[\w\-]*+ ?)/)
+    cuser_groups_length = cuser_groups_array.length
+    for i in 0...cuser_groups_length
+      cuser_groups_array[i].strip! # Strip any leading or trailing characters off the groups captured by the regex.
+    end
+
+    # Remove erronious entries that don't belong here but are captured by the regex.
+    cuser_groups_array.delete('GROUP INFORMATION')
+    cuser_groups_array.delete('Group Name')
+    cuser_groups_array
+  end
+
+  #
+  # See if +path+ on the remote system is accessible in the manner specified by +perm+ by the current user.
+  #
+  # @param path [String] Remote path to check
+  # @param perms [Array] Array of permissions to check for
+  #
+  # @return [Boolean] true if +path+ is accessible to the current user via any of the manners listed in +perms+, otherwise false.
+  def check_win_path_permissions(path, perms)
+    # Next check if icacls exists as a command on the target system, and switch to calcs if not.
+    if command_exists?('icacls')
+      # Grab permissions of the path in question and extract relevant info using a regex.
+      # Also strip extra whitespace before comparison occurs.
+      icacls_output_raw = cmd_exec("icacls \"#{path}\"").to_s
```

Comment:
```suggestion
```

Reply:
Removed and updated appropriately in https://github.com/rapid7/metasploit-framework/pull/15987/commits/21be7a0ee8a0de1cc3ab01a8d117819122a89120